### DISCUSSION TITLE
Udostępnienie ACR dla innych subskrypcji.

### DISCUSSION THREAD

**Discussion by Maniek666** [AzureCommunityPL/group#140] [Pomoc] [answered]
Hej,

Mam 3 subskrypcje, gdzie na jednej mam postawiony Azure Container Registry. Czy da radę jakoś go udostępnić do pozostaych dwóch? Chciabym mieć centralne repozytiorium, gdzie moglbym dorzucać ACI z pozostaych dwóch subskrypcji.

Dzięki.

**Comment by wi5nia** [comment]
Ale jakby w czym problem? Do ACRa mozesz pchac z dowolnego miejscia tylko musisz sie uwierzytelnic i tyle.

No chyba, ze nie chces zpublicznie miec tego na swiat w ogole wystawionego to idziesz w private linka albo service endpoint (preview) i odpowiednia infrastrukture i zabezpieczenie sieciowe.

**Comment by Maniek666** [comment]
![aci](https://user-images.githubusercontent.com/72091909/98396173-9b086c80-205d-11eb-8127-a489105143cc.png)

Chodzi mi o to żeby z tych pozostaytch dwóch subskypcji widzial mi ten ACR, jak na screenie wrzuciem. Nie mogę go wybrać, bo istnieje w innej subskybcji.

Dzięki.

**Reply by mgrabarz** [reply]
To tylko GUI, tak to zakodowali w Portalu że nie widać, co nie znaczy że się nie da w kodzie/armie/skrypcie.
Na koniec dnia to tylko podstawia "nazwę" ACRa i jego admin credentials.

Odpowiadając na twoje pytanie: nie da się zobaczyć ACRa z innej subskrypcji w graficznym kreatorze ACI. Możesz zgłosić issue do Portalu by to poprawili.

**Comment by kaluzaaa** [comment, marked as answer]
trzeba wybrać private repo i zadziała jak się poda namiary na acr ;-)

Wysłane z iPada

Wiadomość napisana przez Marek Grabarz <notifications@github.com> w dniu 06.11.2020, o godz. 18:55:

﻿

To tylko GUI, tak to zakodowali w Portalu że nie widać, co nie znaczy że się nie da w kodzie/armie/skrypcie.
Na koniec dnia to tylko podstawia "nazwę" ACRa i jego admin credentials.

Odpowiadając na twoje pytanie: nie da się zobaczyć ACRa z innej subskrypcji w graficznym kreatorze ACI. Możesz zgłosić issue do Portalu by to poprawili.

**Comment by Maniek666** [comment]
Dzięki Panowie za potweirdzenie, myślalem, że jakieś ustawienie mi umknęlo. Zgloszę to na Portal, może kiedyś dorzucą :)

Miego weekendu i na pewno do spisania :)

**Comment by Maniek666** [comment]
Hej, 

Udao mi się faktycznie za pomocą ARMa dobić do ACR, ale mam problem żeby podlączyć się po istniejącego vnet'a i subnet'a. 
Dostaje bląd, że vnet jest nie zdefiniowany:
Deployment template validation failed: 'The resource 'Microsoft.Network/virtualNetworks/myVnet. ' is not defined in the template.
[aci.txt](https://github.com/AzureCommunityPL/group/files/5524425/aci.txt)

Podpowie mi ktoś jak to zdefiniować? 

Dzięki.

**Reply by mgrabarz** [reply]
Nie można robić DependsOn do zasobów tworzonych poza szablonem. W tym przypadku masz dwie opcje:
- tworzyć sieć w szablonie obok aci,
- usunąć DependsOn VNet i założyć że sieć już stoi na etapie odpalenia szablonu aci.

**Comment by Maniek666** [comment]
Dzięki, usunąlem dodalem jeszcze niezbędne rzeczy, żeby robil się w innej resource groupie niż w której jest vnet. 

Dziala jak ta lala.

Dzięki!
